datafusion/physical-expr-common/src/aggregate/groups_accumulator/prim_op.rs

```diff
@@ -134,6 +134,46 @@ where
         self.update_batch(values, group_indices, opt_filter, total_num_groups)
     }
 
+    fn convert_to_state(
+        &self,
+        values: &[ArrayRef],
+        opt_filter: Option<&BooleanArray>,
+    ) -> Result<Vec<ArrayRef>> {
+        let values = values[0].as_primitive::<T>();
+        let mut state = PrimitiveBuilder::<T>::with_capacity(values.len())
+            .with_data_type(self.data_type.clone());
+
+        match opt_filter {
+            Some(filter) => {
+                values
```

Comment:
Can use filter kernel here instead of zipping?

Reply:
Not sure about filter, but kernels sound like a good idea, I'll try to switch to using them.

Reply:
Yes, but the example shows that it filters out values from the source array, and conversion to state must produce the same number of elements, just placing nulls instead of filtered values, so I'm planning to look for smth like "apply null mask".

I've started with some benchmarks (criterion based ones) and they show that current code for nullable columns (at least for count) is significantly slower that for non nullable ones (~15 times :disappointed: ), probably some part of this time can be recovered.

Reply:
For "no filter" -- casting `values.logical_nulls()` to i64 helps a bit. Regarding `bitwise_and` -- I'll try (the problem with all logical functions is that `filter` may also contain nulls)